src/std/misc/list.ss

```diff
@@ -161,13 +161,12 @@
 (defrules with-list-builder ()
   ((_ (c) body1 body+ ...) (with-list-builder (c _) body1 body+ ...))
   ((_ (poke peek) body1 body+ ...)
-   (let ((head (cons #f '()))) ;; use a traditional implementation of queue as cons of tail and head
-     (set-car! head head)
+   (let* ((head (cons #f '())) ;; use a traditional implementation of queue as cons of tail and head
+          (tail head))
      (defrules poke ()
-       ((_ val) (let ((old-tail (car head))
-                      (new-tail (cons val '())))
-                  (set-cdr! old-tail new-tail)
-                  (set-car! head new-tail)))
+       ((_ val) (let ((new-tail (cons val '())))
+                  (set-cdr! tail new-tail)
```

Comment:
we might want to use `##set-cdr!` here, so that always have an unchecked mutation.